### PR TITLE
XPlatCppSdk Bugfixes and test additions.

### PR DESCRIPTION
**Pull request by pgilmorepf**
Http.Update will report the correct number of active API calls.
Numbers in json will [de]serialize correctly again.
Fixed timestamp [de]serialization in Windows.
Fixed some resharper complaints.

### CODE REVIEW

File: targets/XPlatCppSdk/source/cppsdk/include/playfab/PlayFabBaseModel.h [hunk 1]
```diff
@@ -99,12 +99,10 @@ namespace PlayFab
     inline void FromJsonUtilT(const Json::Value& input, time_t& output)
     {
         if (input == Json::Value::null) return;
-        std::string enumStr = input.asString();
+        const std::string timeStr = input.asString();
         tm timeStruct = {};
-        std::istringstream iss(enumStr.c_str());
+        std::istringstream iss(timeStr);
         iss >> std::get_time(&timeStruct, "%Y-%m-%dT%T");
```
Comment (debhaldarMS) [inline targets/XPlatCppSdk/source/cppsdk/include/playfab/PlayFabBaseModel.h]: Should we check what happens if the parse of "iss" fails ? Can this bubble up as an exception in the game ?

File: targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp [hunk 2]
```diff
@@ -226,20 +229,23 @@ namespace PlayFab
...
             reqContainer = pendingResults[pendingResults.size() - 1];
             pendingResults.pop_back();
+            activeRequestCount--;
         } // UNLOCK httpRequestMutex
 
         HandleResults(*reqContainer);
         delete reqContainer;
-        return resultCount;
+
+        // activeRequestCount can be altered by HandleResults, so we have to re-lock and return an updated value
```
Comment (debhaldarMS) [inline targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp]: I'm not seeing how HandleResults can modify activeRequestsCount - can you please help me understand ?

File: targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp [hunk 3]
```diff
@@ -226,20 +229,23 @@ namespace PlayFab
...
+            if (pendingResults.empty())
+                return activeRequestCount;
 
             reqContainer = pendingResults[pendingResults.size() - 1];
             pendingResults.pop_back();
+            activeRequestCount--;
         } // UNLOCK httpRequestMutex
 
         HandleResults(*reqContainer);
         delete reqContainer;
```
[the remark this answers is not in the document]
Comment (pgilmorepf) [reply targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp]: reqContainer is entirely internal. ONLY the PlayFabHttp class touches it. It's never passed outside of PlayFabHttp. reqContainer contains the PlayFab result object, which is the only thing sent to the customer's code.

File: targets/XPlatCppSdk/source/cppsdk/include/playfab/PlayFabBaseModel.h (hunk 1 above)
Comment (pgilmorepf) [reply targets/XPlatCppSdk/source/cppsdk/include/playfab/PlayFabBaseModel.h]: I think this is a larger discussion. The entire serialization scheme basically tries its best, and defaults to something (hopefully reasonable) if it fails. I'm open to discussing the proposal, but we would need to handle it in ALL deserializations, not just this one. It's certainly not a trivial change, and shouldn't be in this checkin.

File: targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp (hunk 2 above)
Comment (pgilmorepf) [reply targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp]: HandleResults invokes customer code.
Customer code can do anything, including start another API call.

File: targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (debhaldarMS) [reply targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp]: As discussed, let's log a bug for this!

File: targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp (hunk 2 above)
Comment (debhaldarMS) [reply targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp]: agreed

**Pull request merged**